openpype/hosts/tvpaint/plugins/publish/extract_sequence.py

```diff
@@ -346,12 +459,50 @@ def render(
         )
         # Render layers
         filepaths_by_layer_id = {}
+        opacity_by_layer_id = {}
+
+        # in case some layers are visible, but nothing is rendered,
+        # for example, if the post behaviour is on none, and the layer exposure frames
+        # are not in the custom_mark_range
+        output_used_layers_id = []
         for layer_id, render_data in extraction_data_by_layer_id.items():
             layer = layers_by_id[layer_id]
+            output_used_layers_id.append(layer_id)
             filepaths_by_layer_id[layer_id] = self._render_layer(
-                render_data, layer, output_dir, ignore_layer_opacity
+                render_data, layer, output_dir, custom_mark_range
             )
 
+            opacity_by_layer_id[layer_id] = self._convert_opacity_range(layer["opacity"])
+
+        # filter the layers to compose to remove those who didn't have a render
+        filtered_layers = [layer for layer in layers if layer['layer_id'] in output_used_layers_id]
+
+        # Prepare bg image if apply_background enabled
+        if apply_background:
+            # Create and add to Layers
+            bg_layer = {}
+            layer_positions = []
+            layer_ids = []
+            for layer in layers:
+                layer_positions.append(layer["position"])
+                layer_ids.append(layer["layer_id"])
+
+            bg_position = max(layer_positions)+1
+            bg_id = max(layer_ids)+1
+
+            bg_layer["frame_end"] = mark_out
+            bg_layer["frame_start"] = mark_in
+            bg_layer["layer_id"] = bg_id
+            bg_layer["position"] = bg_position
+            bg_layer["opacity"] = 100
+            bg_layer["name"] = "BG_OP_Layer"
+
+            filtered_layers.append(bg_layer)
+
+            # Add to extraction_data_by_layer_id
+            filepaths_by_layer_id[bg_id] = self._create_bg_images(bg_layer, output_dir, resolution, custom_mark_range)
+            opacity_by_layer_id[bg_id] = self._convert_opacity_range(100)
```

Comment:
Why not simply put `255`

Reply:
Indeed, it may come from copy paste some part of the code